bin/ci

```diff
@@ -74,7 +74,28 @@ prepare_system() {
 
 build() {
   with_build_env 'make std_spec clean threads=1'
-  with_build_env 'make crystal std_spec compiler_spec docs threads=1'
+
+  case $ARCH in
+    i386)
+      with_build_env 'make crystal threads=1'
+      with_build_env 'SPEC_SPLIT="0%4" make std_spec threads=1'
+      with_build_env 'SPEC_SPLIT="1%4" make std_spec threads=1'
+      with_build_env 'SPEC_SPLIT="2%4" make std_spec threads=1'
+      with_build_env 'SPEC_SPLIT="3%4" make std_spec threads=1'
+
+      parts=16
+      for i in $(seq $parts)
+      do
+        with_build_env "CRYSTAL_SPEC_COMPILER_THREADS=1 SPEC_SPLIT=\"$((i-1))%$parts\" make compiler_spec threads=1"
```

Comment:
That's strange, a while loop would be clearer (no need to use seq nor `i -1`)

Reply:
my bash-fu is limited.

Reply:
A classic counter:
```
i=0
while [ $i -lt $parts ]; do
  ...
  i=$((i + 1))
done
```

Anyway, wouldn't it be better to have this logic in the makefile, in order to make it more generic and use it outside the CI context?

Reply:
seq replaced by while.

The whole goal of this is for the CI. I don't spec the index filter to become a public API.